### PR TITLE
chore(migration): update message

### PR DESCRIPTION
**Pull request by xzf0587**

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -146,7 +146,7 @@
...
   "core.aadManifestMigration.ParsePermissionsFailedWarning": "Cannot parse permissions.json file when migration, and please follow this link to update permissions in aad.template.json file if necessary: https://aka.ms/teamsfx-aad-manifest#customize-requiredresourceaccess",
-  "core.projectVersionChecker.vscodeUseNewVersion": "Your Teams Toolkit extension no longer supports the current project, please upgrade the extension to the latest version and try to open this project again.",
+  "core.projectVersionChecker.vscodeUseNewVersion": "The current project is incompatible with the installed Teams Toolkit, please install %s or above and reopen this project.",
   "core.projectVersionChecker.cliUseNewVersion": "[core] Your TeamsFx CLI is too old to support current project, please upgrade to the latest version using command below:\nnpm install -g @microsoft/teamsfx-cli@latest",
```
Comment (blackchoey) [inline packages/fx-core/resource/package.nls.json:150]: do we need to update the CLI message to tell users which version to install?

File: packages/fx-core/src/core/middleware/projectVersionChecker.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ async function showDialog(ctx: CoreHookContext) {
   if (inputs.platform === Platform.VSCode) {
     await TOOLS?.ui.showMessage(
       "warn",
-      getLocalizedString("core.projectVersionChecker.vscodeUseNewVersion"),
+      getLocalizedString("core.projectVersionChecker.vscodeUseNewVersion", "Teams Toolkit 5.0.0"),
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/projectVersionChecker.ts:49]: It's may be hard to determine which version to install. For example, maybe one day Teams Toolkit 6.0 bumps the version in teamsapp.yml to 2.0.0, then the hint "Teams Toolkit 5.0.0" would be incorrect when user use old Teams Toolkit.

Should we just ask users install latest toolkit and provide an aka link here?

Comment (xzf0587) [reply packages/fx-core/src/core/middleware/projectVersionChecker.ts:49]: The message has been discussed with @jidddddd.
Actually, the message just suggests a lowest toolkit version
image

**Review by jidddddd — DISMISSED** [review]

**Review by sffamily — APPROVED** [review]

File: packages/fx-core/src/core/middleware/projectVersionChecker.ts [hunk 3]
```diff
@@ -44,21 +46,30 @@ async function showDialog(ctx: CoreHookContext) {
...
   } else if (inputs.platform === Platform.CLI) {
     TOOLS?.logProvider.warning(getLocalizedString("core.projectVersionChecker.cliUseNewVersion"));
   } else if (inputs.platform === Platform.VS) {
-    await TOOLS?.ui.showMessage(
+    const res = await TOOLS?.ui.showMessage(
       "warn",
-      getLocalizedString("core.projectVersionChecker.vscodeUseNewVersion"),
+      getLocalizedString(
+        "core.projectVersionChecker.vscodeUseNewVersion",
+        "Visual Studio 2022 17.5 Preview"
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/projectVersionChecker.ts:65]: Should it be Visual Studio 2022 17.5 Preview 3?

File: packages/fx-core/src/core/middleware/projectVersionChecker.ts [hunk 4]
```diff
@@ -44,21 +46,30 @@ async function showDialog(ctx: CoreHookContext) {
   const lastArg = ctx.arguments[ctx.arguments.length - 1];
   const inputs: Inputs = lastArg === ctx ? ctx.arguments[ctx.arguments.length - 2] : lastArg;
   if (inputs.platform === Platform.VSCode) {
-    await TOOLS?.ui.showMessage(
+    const res = await TOOLS?.ui.showMessage(
       "warn",
-      getLocalizedString("core.projectVersionChecker.vscodeUseNewVersion"),
+      getLocalizedString("core.projectVersionChecker.vscodeUseNewVersion", "Teams Toolkit 5.0.0"),
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/projectVersionChecker.ts:51]: Teams Toolkit 5.0 does not exist in marketplace stable release. Should we call it Teams Toolkit 5.0 Pre-Release version?

File: packages/fx-core/src/core/middleware/projectVersionChecker.ts (hunk 3 above)
Comment (xzf0587) [reply packages/fx-core/src/core/middleware/projectVersionChecker.ts:65]: The version message " Visual Studio 2022 17.5 Preview 3 or above" does not look very well.
The version above a preview may be confused for user.

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**